consensus/scheme/rolldpos/rolldposctx.go

```diff
@@ -484,6 +485,14 @@ func (ctx *rollDPoSCtx) Commit(msg interface{}) (bool, error) {
 	switch err := ctx.chain.CommitBlock(pendingBlock); errors.Cause(err) {
 	case blockchain.ErrInvalidTipHeight:
 		return true, nil
+	case factory.ErrNotValidated:
+		// if the cache has been reset, we need to revalidate/commit the block
+		if err := ctx.chain.ValidateBlock(pendingBlock); err != nil {
+			return false, err
+		}
+		if err := ctx.chain.CommitBlock(pendingBlock); err != nil {
+			return false, err
+		}
```

Comment:
as discussed offline, these code won't be hit.

Reply:
yes! moved error handling logic into blockchain.go commitblock()